### ISSUE TITLE
Improve error reporting with fix suggestions

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#12]
## Problem

Current error reporting shows violations but doesn't provide actionable fix suggestions to help users resolve issues quickly.

## Proposed Solution

Enhance violation reporting to include suggested fixes where possible.

### Implementation Areas

1. **Violation Structure**
   - Add optional fix suggestion field to Violation struct
   - Include before/after examples when applicable
   - Provide specific actionable guidance

2. **Rule Enhancement**
   - Update existing rules to include fix suggestions
   - Add helper methods for common fix patterns
   - Ensure suggestions are contextually appropriate

3. **Output Formatting**
   - Display fix suggestions in CLI output
   - Format suggestions clearly and consistently
   - Consider different output formats (human vs machine readable)

## Acceptance Criteria

- [ ] Violation struct supports optional fix suggestions
- [ ] At least 3 existing rules provide fix suggestions
- [ ] CLI output displays suggestions clearly
- [ ] Fix suggestions are tested
- [ ] Documentation explains suggestion format

**Comment by joshrotenberg** [comment]
## Infrastructure Status Update

✅ **Fix suggestion infrastructure is fully implemented!**

### What's Already Available
- ✅ **`Fix` struct** with description, replacement text, and precise position info
- ✅ **`Violation` struct** includes optional `fix: Option<Fix>` field  
- ✅ **Position tracking** with line/column precision for replacements
- ✅ **Serialization support** for JSON/other output formats
- ✅ **Complete API** for creating and handling fix suggestions

### Current Status
The **infrastructure is complete** but no rules are currently providing fix suggestions. The missing piece is updating individual rules to populate the `fix` field.

### Implementation Ready
Rules can now easily provide fixes like:
```rust
Violation {
    // ... other fields
    fix: Some(Fix {
        description: "Remove trailing spaces".to_string(),
        replacement: Some("\n".to_string()), // or None for deletion
        start: Position { line: 1, column: 5 },
        end: Position { line: 1, column: 8 },
    }),
}
```

### Next Steps
The infrastructure is done - this issue now just needs individual rules to be updated to provide fix suggestions. This could be tackled rule-by-rule as enhancements.

**Comment by joshrotenberg** [comment]
## Correction: Fix Support IS Implemented! 

✅ **Fix suggestions are already working for multiple rules!**

### Current Implementation (from PR #99)
We already have working fix support for:

- ✅ **MD010 (no-hard-tabs)**: Replaces tabs with configurable number of spaces
- ✅ **MD047 (single-trailing-newline)**: Adds missing newline or removes extras at EOF

### Infrastructure (Complete)
- ✅ **`Fix` struct** with description, replacement, and position tracking
- ✅ **`Violation` struct** includes `fix: Option<Fix>` field
- ✅ **Position tracking** for precise text replacements
- ✅ **Helper methods**: `create_violation_with_fix()` for easy fix creation

### Example Implementation (MD010)
```rust
let fix = Fix {
    description: format!("Replace tab with {} spaces", self.spaces_per_tab),
    replacement: Some(fixed_line),
    start: Position { line: line_num, column: 1 },
    end: Position { line: line_num, column: line.len() + 1 },
};
```

### Status Assessment
The core requirement is **already met** - we have:
- ✅ Violation struct supports fix suggestions
- ✅ 2+ rules provide fix suggestions (MD010, MD047)
- ✅ Fix suggestions are tested and working

### Potential Enhancements
While the core feature is complete, we could:
- Add fix support to more rules (MD009 trailing spaces, MD012 blank lines, etc.)
- Enhance CLI output formatting for fix display
- Add `--fix` flag to auto-apply suggested fixes

**This issue could be closed as complete**, or kept open for expanding fix support to more rules.

**Comment by joshrotenberg** [comment]
Closing as complete. Fix suggestions are extensively implemented across 40+ rules. The infrastructure is in place with the Fix struct, Violation with optional fix field, and helper methods. The --fix flag is also implemented for auto-applying fixes.

Current state:
- Fix struct with description, replacement, and position tracking
- create_violation_with_fix() helper for easy fix creation  
- 40+ rules have fix support (MD001-MD050 range)
- CLI --fix flag applies fixes automatically
- Tests cover fix functionality

Any further expansion of fix support to additional rules can be tracked separately.
